src/services/BaseService.js

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import utils from '../utils';
+
+function makeHttpRequest(method, uri, body, headers, callback) {
+  var request = new XMLHttpRequest();
+
+  request.open(method.toUpperCase(), uri, true);
+
+  if (headers) {
+    for (var name in headers) {
+      var value = headers[name];
+      request.setRequestHeader(name, value);
+    }
+  }
+
+  request.onreadystatechange = function () {
+    // 4 = Request finished and response is ready.
+    // Ignore everything else.
+    if (request.readyState !== 4) {
+      return;
+    }
+
+    var result = {
+      status: request.status,
+      responseJSON: null
+    };
+
+    try {
+      if (request.responseText) {
+        result.responseJSON = JSON.parse(request.responseText);
+      }
+      callback(null, result);
+    } catch(e) {
+      callback(e);
+    }
+  };
+
+  if (body !== undefined) {
+    request.setRequestHeader('Content-Type', 'application/json; charset=utf-8');
+    request.send(JSON.stringify(body));
+  }
+}
+
+export default class BaseService {
+  constructor(endpoints) {
+    var defaultEndpoints = {
+      baseUri: null
+    };
+
+    this.endpoints = utils.mergeObjects(defaultEndpoints, endpoints);
+  }
+
+  _makeRequest(method, path, body, callback) {
+    var uri = this._buildEndpoint(path);
+
+    var headers = {
+      'Accept': 'application/json'
+    };
+
+    // Only set the X-Stormpath-Agent header if we're on the same domain as the requested URI.
+    // This because we want to avoid CORS requests that require you to have to whitelist the X-Stormpath-Agent header.
+    if (!utils.isSameHost(uri, window.location.href)) {
```

Comment:
I think you should remove the `!`, right? Otherwise this will add the header if they're not on the same host.